Decred Wallet/Features/Navigation Menu/NavigationMenuTabBarController.swift

```diff
@@ -0,0 +1,105 @@
+        super.viewDidLoad()
+        self.loadTabBar()
+        self.floatingButtons = NavMenuFloatingButtons()
+        if self.isNewWallet {
+            self.showOkAlert(message: LocalizedStrings.newWalletMsg)
+        }
+    }
+    
+    func loadTabBar() {
+        let tabItems: [MenuItem] = [.overview, .transactions, .wallets, .more]
+        
+        self.setupCustomTabMenu(tabItems)
+        self.selectedIndex = 0
+    }
+    
+    public func loadFloatingButtons() {
+        self.floatingButtons.translatesAutoresizingMaskIntoConstraints = false
+        self.floatingButtons.clipsToBounds = true
+        self.floatingButtons.layer.shadowColor = UIColor(red: 0.16, green: 0.44, blue: 1.0, alpha: 1).cgColor
+        self.floatingButtons.layer.shadowOpacity = 0.16
+        self.floatingButtons.layer.shadowOffset = CGSize(width: -1, height: 1)
+        self.floatingButtons.layer.masksToBounds = false
+        
+        self.view.addSubview(self.floatingButtons)
+        let constraints = [
+            self.floatingButtons.heightAnchor.constraint(equalToConstant: 48), // Fixed height for floating buttons
+            self.floatingButtons.widthAnchor.constraint(equalToConstant: 240), // Fixed width for floating buttons
+            self.floatingButtons.centerXAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.centerXAnchor),
+            self.floatingButtons.bottomAnchor.constraint(equalTo: self.customTabBar.topAnchor, constant: -12), // position floating buttons 12pts above the nav menu
+        ]
+        NSLayoutConstraint.activate(constraints)
+    }
+    
+    // Create our custom menu bar and display it right where the tab bar should be.
+    func setupCustomTabMenu(_ menuItems: [MenuItem]) {
+        let frame = tabBar.frame
+        var controllers = [UIViewController]()
+        
+        self.customTabBar = CustomTabMenuView(items: menuItems, frame: frame) // Draw and layout the tab navigation menu
+        
+        self.view.addSubview(customTabBar)
+        self.customTabBar.translatesAutoresizingMaskIntoConstraints = false // We are setting positioning constraints in the next line. best to ignore XCode generated constraints
+        self.customTabBar.clipsToBounds = true
+        
+        // Positioning constraints to place the nav menu right where the tab bar should be
+        NSLayoutConstraint.activate([
+            self.customTabBar.leadingAnchor.constraint(equalTo: tabBar.leadingAnchor),
+            self.customTabBar.trailingAnchor.constraint(equalTo: tabBar.trailingAnchor),
+            self.customTabBar.widthAnchor.constraint(equalToConstant: tabBar.frame.width),
+            self.customTabBar.heightAnchor.constraint(equalToConstant: 67), // Fixed height of 67pts for nav menu. This value does not include the curved edge insets in devices >= iPhone X
+            self.customTabBar.bottomAnchor.constraint(equalTo: tabBar.bottomAnchor)
+        ])
+        
+        for i in 0 ..< menuItems.count {
```

Comment:
The declaration and initialization of `controllers` and this for loop can be moved to the end of this function, closer to where the array is actually used.

Reply:
resolved in f116bc6